src/printing/mod.rs

```diff
@@ -0,0 +1,44 @@
+use quote::{ToTokens, TokenStreamExt};
+
+use crate::Buffer;
+
+macro_rules! to_tokens_simple {
+    ($id:ident) => {
+        impl<B: Buffer + Clone> ToTokens for crate::$id<B> {
+            fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
+                tokens.append(<Self as Into<proc_macro2::Literal>>::into(self.clone()))
```

Comment:
That shouldn't require a clone. I suppose there should be `From<&Literal> for proc_macro2::Literal` impls hu? The only thing the current impl (for non references) does is call `parse` on the `raw_input()`. That works with a reference as well. Can you add these impls in a separate commit? And then you can use them here and remove the `Clone` bound. If you prefer I can also add those impls.

Also, `tokens.append(proc_macro2::Literal::from(self))` should work as well?